src/uri/authority.rs

```diff
@@ -192,6 +192,64 @@ impl PartialEq<str> for Authority {
     }
 }
 
+impl PartialEq<Authority> for str {
+    fn eq(&self, other: &Authority) -> bool {
+        self.eq_ignore_ascii_case(other.as_str())
+    }
+}
+
+impl PartialEq<String> for Authority {
+    fn eq(&self, other: &String) -> bool {
+        self.data.eq_ignore_ascii_case(other.as_str())
+    }
+}
+
+impl PartialEq<Authority> for String {
+    fn eq(&self, other: &Authority) -> bool {
+        self.as_str().eq_ignore_ascii_case(other.as_str())
+    }
+}
+
+/// Case-insensitive ordering
+///
+/// # Examples
+///
+/// ```
+/// # use http::uri::Authority;
+/// let authority: Authority = "DEF.com".parse().unwrap();
+/// assert!(authority < *"ghi.com");
+/// assert!(authority > *"abc.com");
```

Comment:
Curious, is the deref needed? Is that because there is an impl for `str`, but not `&'a str`?

Reply:
I copied the other doc test here for `PartialEq`
I did find that either the str needs to deref or authority need to be a ref. Perhaps there's a better way? (I'm a little new to rust so I'm still working through some of these details.)

Reply:
I believe if you add an implementation of `PartialEq<&'a str> for Authority` and `PartialEq<Authority> for &'a str`, then it can be written without the derefs.